File: amprobe/service/model/alarm.go
// Package model
// Date:   2024/10/14 17:21
// Author: Amu
// Description:
package model

import "gorm.io/gorm"

type AlarmThreshold struct {
	gorm.Model
	Type      string `gorm:"type:varchar(255)comment:告警类型"`
	Duration  int    `gorm:"comment:取值间隔"`
	Threshold int    `gorm:"comment:阈值"`
}

func (d *AlarmThreshold) TableName() string {
	return "s_alarm_threshold"
}

File: amprobe/web/src/interface/alarm.ts
/**
 * @Author     : Amu
 * @Date       : 2024/10/17 16:35:32
 * @Description:
 */

export interface EmailSetting {
    id: number
    server: string
    port: number
    sender: string
    password: string
    receiver: string
}

export interface AlarmThreshold {
    id: number
    type: string
    duration: number
    threshold: number
}

export interface AlarmThresholdQueryResult {
    data: AlarmThreshold[]
}

export interface AlarmThresholdUpdateArgs {
    id: number
    type: string
    duration: number
    threshold: number
}
